### ISSUE TITLE
Parse error in multiline subscription expression

### ISSUE THREAD

**Issue by Scony** [godotengine/godot#35417]
**Godot version:**
master @ d11d7dfe3e15e34ba400e6b73d618abc4b8f5886

**OS/device including version:**
Linux

**Issue description:**
Godot raises parse error on the code below:
```
class X:
	func foo(p):
		var x = p[
			1
		]
```
```
Godot Engine v3.2.rc.custom_build.d11d7dfe3 - https://godotengine.org
 
SCRIPT ERROR: GDScript::reload: Parse Error: Error parsing expression, misplaced: '\n'
   At: res://tests/potential-godot-bugs/multiline-subscription-expression.gd:4.
ERROR: reload: Method/Function Failed, returning: ERR_PARSE_ERROR
   At: modules/gdscript/gdscript.cpp:576.
```

**Comment by bojidar-bg** [comment]
Adding `parenthesis++` and `parenthesis--` around this line should resolve the issue:
https://github.com/godotengine/godot/blob/1c6d1fcf956b882fc8a14fa595b7f1e9b15e1574/modules/gdscript/gdscript_parser.cpp#L1251-L1253

Note that the following code would continue erroring, but it should be resolved with #35415:
```gdscript
var x = (z
    [p])
```

**Comment by RandomShaper** [comment]
I don't think that should be changed. The language is just designed after Python in that regard: to require parentheses for allowing line breaks in expressions).

(See https://github.com/godotengine/godot/issues/208#issuecomment-39650721 for context.)

**Comment by lakshay-angrish** [comment]
@RandomShaper 
Shouldn't the newline after `[` be ignored by the following code?
https://github.com/godotengine/godot/blob/a61f5a75209c69e617f5d9ac8f74c970077b931a/modules/gdscript/gdscript_parser.cpp#L946-L988

**Comment by RandomShaper** [comment]
I think that's for the context when an array is declared, to allow one item per line:
```
var a = [
    1,
    2,
    3,
]
```

What is happening in the example code is indexing, which is parsed here: https://github.com/godotengine/godot/blob/a61f5a75209c69e617f5d9ac8f74c970077b931a/modules/gdscript/gdscript_parser.cpp#L1245

**Comment by bojidar-bg** [comment]
> I don't think that should be changed. The language is just designed after Python in that regard: to require parentheses for allowing line breaks in expressions).

`[` and `]` are technically parenthesis.
I mean, the following two are identical:
```
p[(x + y)]
p[x + y]
```
So it makes sense that the following two are:
```
p[(
  x + y
)]

p[
  x + y
]
```

**Comment by RandomShaper** [comment]
I wasn't too convinced about your point since I thought the rule was intended for actual parentheses, rather than expression "enclosing".

I was wrong.

Moreover, regardless any other consideration, since Python allows it and the intent is that GDScript imitates its syntax, it should be allowed.

**Comment by Vignesh1-art** [comment]
This issue doesn't seem to be there in master branch.

**Comment by dalexeev** [comment]
Fixed by #54227.
